packages/console/src/pages/SignInExperience/components/ManageLanguageModal/AddLanguageSelector.tsx

```diff
@@ -0,0 +1,60 @@
+import { languages, LanguageTag } from '@logto/language-kit';
+import { useRef, useState } from 'react';
+
+import Button from '@/components/Button';
+import Dropdown, { DropdownItem } from '@/components/Dropdown';
+import Plus from '@/icons/Plus';
+
+import * as style from './AddLanguageSelector.module.scss';
+
+type Props = {
+  options: LanguageTag[];
+  onSelect: (languageTag: LanguageTag) => void;
+};
+
+// TODO:(LOG-4147) Support Instant Search In Manage Language Editor Dropdown
+const AddLanguageSelector = ({ options, onSelect }: Props) => {
+  const anchorRef = useRef<HTMLDivElement>(null);
+  const [isDropDownOpen, setIsDropDownOpen] = useState(false);
+
+  return (
+    <div>
+      <div ref={anchorRef}>
```

Comment:
Could we apply the ref to the button element it self? Just avoid extra layer of wrapper.

Reply:
The `<Button />` component is a Function component and cannot be given refs unless I make some changes to the `<Button />` component code, but I'm not sure if this modification will cause some potential problems, so I wrapped it with a `<div />`.

Reply:
Prefer to update component using forwardRef instead of introducing a extra container.

Reply:
Add an issue to track this: [LOG-4282](https://linear.app/silverhand/issue/LOG-4282/forward-reference-for-the-button-component-and-refactor-related-code)